components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java

```diff
@@ -158,4 +233,264 @@ public void testDeleteScope() throws Exception {
         oAuth2ScopeService.deleteScope(SCOPE_NAME);
         oAuth2ScopeService.getScope(SCOPE_NAME);
     }
+
+    @Test
+    public void testAddUserConsentForApplication() throws Exception {
+
+        List<String> approvedScopes = new ArrayList<>(Arrays.asList("read", "write"));
+        List<String> deniedScopes = new ArrayList<>(Arrays.asList("delete"));
+        String appId = UUID.randomUUID().toString();
+        insertAppId(appId);
+        oAuth2ScopeService.addUserConsentForApplication("user_id", appId, 1, approvedScopes, deniedScopes);
+
+        OAuth2ScopeConsentResponse oAuth2ScopeConsentResponse = oAuth2ScopeService.getUserConsentForApp("user_id",
+                appId, 1);
+        assertEquals(oAuth2ScopeConsentResponse.getApprovedScopes().get(0), approvedScopes.get(0));
+        assertEquals(oAuth2ScopeConsentResponse.getDeniedScopes().get(0), deniedScopes.get(0));
+    }
+
+    private void insertAppId(String uuid) throws Exception {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection()) {
+            String sql = "INSERT INTO SP_APP (TENANT_ID, APP_NAME, UUID) VALUES (?,?,?)";
+            PreparedStatement ps = connection.prepareStatement(sql);
```

Comment:
Prepared statements and result sets (if any) should be explicitly closed as well for proper cleanup. Please add this to the `try-with-resources` statement.

Reply:
Addressed with commit [9e575a3](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1691/commits/9e575a323c0604e9985d430a5b3fee4942d86cb1)